src/core/layout/qgslayoutitemelevationprofile.cpp

```diff
@@ -401,6 +433,26 @@ bool QgsLayoutItemElevationProfile::readPropertiesFromElement( const QDomElement
   {
     mPlot->readXml( plotElement, context );
   }
+
+  {
+    mLayers.clear();
+    const QDomElement layersElement = itemElem.firstChildElement( QStringLiteral( "layers" ) );
+    QDomElement layerElement = layersElement.firstChildElement( QStringLiteral( "layer" ) );
+    while ( !layerElement.isNull() )
+    {
+      const QString layerId = layerElement.attribute( QStringLiteral( "id" ) );
+      const QString layerName = layerElement.attribute( QStringLiteral( "name" ) );
+      const QString layerSource = layerElement.attribute( QStringLiteral( "source" ) );
+      const QString layerProvider = layerElement.attribute( QStringLiteral( "provider" ) );
+
+      QgsMapLayerRef ref = QgsMapLayerRef( layerId, layerName, layerSource, layerProvider );
```

Comment:
@nyalldawson , we don't have a util function to save / restore a QgsMapLayerRef to / from XML?

Reply:
Good point, I've moved this to the QgsMapLayerRef code

Reply:
@nyalldawson , thanks.